### PR TITLE
Add documentation for triple slash directives

### PR DESCRIPTION
**Pull request by mhegazy**
triple-slash references and co.

### CODE REVIEW

File: pages/Triple-Slash Directives.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+A triple-slash reference path is resolved relative to the containing file, if unrooted.
+
+### Errors
+
+It is an error to reference a file that does not exist.
+It is an error to for a file to have a triple-slash reference to itself.
+
+### Using `--noResolve`
+
+If the compiler flag `--noResolve` is specified, triple-slash references are ignored; they niether result in adding new files, nor change the order of the files provided.
```
Comment (blakeembrey) [inline pages/Triple-Slash Directives.md]: "... neither ..."

File: pages/Triple-Slash Directives.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+These root files are preprocessed in the same order they are specified.
+Before a file is added to the list, all triple-slash references in it are processed, and their targets included.
+Triple-slash references are resolved in a depth first manner, in the order they have been seen in the file.
+
+A triple-slash reference path is resolved relative to the containing file, if unrooted.
+
+### Errors
+
+It is an error to reference a file that does not exist.
+It is an error to for a file to have a triple-slash reference to itself.
```
Comment (blakeembrey) [inline pages/Triple-Slash Directives.md]: "... to for..."

File: pages/Triple-Slash Directives.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+These are single-line comments, as per the language specification, containing a single XML tag.
```
Comment (sandersn) [inline pages/Triple-Slash Directives.md]: 1. These -> Triple-slash directives
2. I'd drop "as per the language specification"

File: pages/Triple-Slash Directives.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+It is an error to for a file to have a triple-slash reference to itself.
+
+### Using `--noResolve`
+
+If the compiler flag `--noResolve` is specified, triple-slash references are ignored; they niether result in adding new files, nor change the order of the files provided.
+
+## `/// <amd-module />`
+
+By default AMD modules are generated anonymous.
+This can lead to problems when other tools are used to process the resulting modules like a bundlers (e.g. `r.js`).
```
Comment (sandersn) [inline pages/Triple-Slash Directives.md]: "the resulting modules, such as bundlers (e.g. `r.js`)."

**Comment by sandersn** [comment]
:+1:

File: pages/Triple-Slash Directives.md (hunk 1 above)
Comment (mhegazy) [inline pages/Triple-Slash Directives.md]: done.

File: pages/Triple-Slash Directives.md (hunk 2 above)
Comment (mhegazy) [inline pages/Triple-Slash Directives.md]: done.

File: pages/Triple-Slash Directives.md (hunk 3 above)
Comment (mhegazy) [inline pages/Triple-Slash Directives.md]: done.

File: pages/Triple-Slash Directives.md (hunk 4 above)
Comment (mhegazy) [inline pages/Triple-Slash Directives.md]: done.

**Pull request merged**